services/app-web/src/common-code/domain-models/submission.ts

```diff
@@ -104,6 +105,24 @@ function submissionName(submission: SubmissionUnionType): string {
     return `MCR-${submission.stateCode.toUpperCase()}-${submission.programIDs.sort(naturalSort).join('-').toUpperCase()}-${padNumber}`
 }
 
+// Pull out the programs names for display from the program IDs
+function programNames(programs: ProgramT[], programIDs: string[]): string[] {
+    return programIDs.map(id => {
+        const program = programs.find(p => p.id === id)
+        if (!program) {
+            return "Unknown Program"
+        }
+        return program.name
+    })
+}
+
+function submissionNameWithPrograms(submission: SubmissionUnionType, statePrograms: ProgramT[]): string {
+    const padNumber = submission.stateNumber.toString().padStart(4, '0')
+    const pNames = programNames(statePrograms, submission.programIDs)
+    const formattedProgramNames = pNames.sort(naturalSort).map(n => n.replace(/\s/g, '-').replace(/[^a-zA-Z0-9+]/g, '').toUpperCase()).join('-')
+    return `MCR-${submission.stateCode.toUpperCase()}-${formattedProgramNames}-${padNumber}`
+}
+
```

Comment:
Can you write some quick unit tests for this second function if it doesn't exist. We have some more involved logic here that would be good for testing